src/main/java/io/reactivex/Single.java

```diff
@@ -874,7 +874,7 @@
      * @return the new Single instance
      * @since 2.0
      */
-    public static <T, R> Single<R> zip(final Iterable<? extends SingleSource<? extends T>> sources, Function<? super Object[], ? extends R> zipper) {
+    public static <T, R> Single<? extends R> zip(final Iterable<? extends SingleSource<? extends T>> sources, Function<? super Object[], ? extends R> zipper) {
```

Comment:
We never return `? extends X` as it adds burden to the receiver. Why did you do this?